lib/diagnostic.js

```diff
@@ -22,13 +33,29 @@ const uniqueWords = function (/* */) {
 // a second argument.
 
 const wordCount = function (/* */) {
-  /* your response here */
+  let wordCount = [];
+
+  let keys = [];
+  let k = 0;
+
+  for(keys[k++] in uniqueWordsAsKeys);
+
+  uniqueWords = keys;
 };
 
 // 4. Write a function that takes a string and returns a dictionary with unique
 // words as keys and a count of each word as the value for that key
-const wordFrequencies = function (/* */) {
-  /* your response here */
+const wordFrequencies = function (freq) {
+  let wordFrequencies = {};
+
+  const addOne = function(num){
+    return (num || 0) +1;
+  };
+
+    for (let i = 0; i < normalizeWords.length; i++) {
+        let word = normalizeWords[i];
+        freq[word] = addOne(freq[word]);
+    }
 };
```

Comment:
You are making the same error here. JavaScript does not magically know what anything is unless it is defined, and you cannot use the same variable name on 2 different things. Please read the solution and make sure you can understand it.
```
const wordFrequencies = function (text) {
  let wordDictionary = {};
  let normalizedWords = normalizeWords(text);
  for (let i = 0; i < normalizedWords.length; i++) {
    let word = normalizedWords[i];
    wordDictionary[word] = addOne(wordDictionary[word]);
  }

  return wordDictionary;
};
```